File: src/main/java/com/sima/leetcode/L824_GoatLatin.java
package com.sima.leetcode;

/**
 * Created by qisima on 4/30/2019 2:58 PM
 */
public class L824_GoatLatin {
    public String toGoatLatin(String S) {
        String[] arr = S.split(" ");
        StringBuilder res = new StringBuilder();
        for (int i = 0; i < arr.length; i++) {
            res.append(toGoatLatinWord(arr[i], i));
        }
        return res.toString();
    }

    private String toGoatLatinWord(String word, int idx){
        StringBuilder sb = new StringBuilder();
        if (idx != 0){
            sb.append(" ");
        }
        if (isVowel(word.charAt(0))){
            sb.append(word).append("ma");
        } else {
            sb.append(word.substring(1)).append(word.charAt(0));
        }
        for (int i = 0; i <= idx ; i++) {
            sb.append('a');
        }
        return sb.toString();
    }

    private Boolean isVowel(char ch) {
        switch (ch){
            case 'A':
            case 'a':
            case 'E':
            case 'e':
            case 'I':
            case 'i':
            case 'O':
            case 'o':
            case 'U':
            case 'u':
                return true;
            default:
                return false;
        }
    }

}

File: src/main/scala/com/sima/L824_GoatLatin.scala
package com.sima

import scala.collection.immutable.HashSet

/**
  * Created by qisima on 4/30/2019 3:25 PM
  */
object L824_GoatLatin {
  def toGoatLatin(S: String): String = {
    val arr = Array('A', 'E', 'I', 'O', 'U', 'a', 'e', 'i', 'o', 'u')
    val sb = new StringBuilder()

    for (word <- S.split(" ").zipWithIndex){
      sb.append(" ")
      if (arr.contains(word._1(0))){
        sb.append(word._1).append("ma")
      }else{
        sb.append(word._1.substring(1)).append(word._1(0)).append("ma")
      }
      (0 to word._2).foreach(sb.append('a'))
    }
    sb.substring(1)
  }
}
